moby fails to build for 32bit x86 on v29.0.0

### Description

The error message is like below:
cmd/docker-proxy/main_linux.go:137:28: undefined: sctp.FileListener

I checked the vendor/github.com/ishidawataru/sctp/sctp_unsupported.go, and it starts with:
// +build !linux linux,386

So this means moby no longer builds for 32bit x86, right?
Is this a deliberate action? Or is it a regression?

### Reproduce

Build moby for 32bit x86 on v29.0.0.

### Expected behavior

_No response_

### docker version

```bash
29.0.0
```

### docker info

```bash
It does not build.
```

### Additional Info

_No response_

--- a/go.mod
+++ b/go.mod
@@ -52,7 +52,7 @@ require (
 	github.com/hashicorp/memberlist v0.4.0
 	github.com/hashicorp/serf v0.8.5
 	github.com/in-toto/in-toto-golang v0.9.0
-	github.com/ishidawataru/sctp v0.0.0-20250829011129-4b890084db30
+	github.com/ishidawataru/sctp v0.0.0-20251114114122-19ddcbc6aae2
 	github.com/miekg/dns v1.1.66
 	github.com/mistifyio/go-zfs/v3 v3.0.1
 	github.com/mitchellh/copystructure v1.2.0

